x/virtualgroup/keeper/keeper.go

```diff
@@ -433,3 +431,110 @@ func (k Keeper) GetGlobalVirtualGroupIfAvailable(ctx sdk.Context, gvgID uint32,
 	}
 	return gvg, nil
 }
+
+func (k Keeper) SetSwapOutInfo(ctx sdk.Context, gvgFamilyID uint32, gvgIDs []uint32, spID uint32, successorSPID uint32) error {
+	store := ctx.KVStore(k.storeKey)
+
+	if gvgFamilyID != types.NoSpecifiedFamilyId {
+		key := types.GetSwapOutFamilyKey(gvgFamilyID)
+		found := store.Has(key)
+		if found {
+			return types.ErrSwapOutFailed.Wrapf("SwapOutInfo of this gvg family(ID: %d) already exist", gvgFamilyID)
+
+		}
+		swapOutInfo := &types.SwapOutInfo{
+			Id:            gvgFamilyID,
+			SpId:          spID,
+			SuccessorSpId: successorSPID,
+		}
+
+		bz := k.cdc.MustMarshal(swapOutInfo)
+		store.Set(key, bz)
+	} else {
+		for _, gvgID := range gvgIDs {
+			key := types.GetSwapOutGVGKey(gvgID)
+			found := store.Has(key)
+			if found {
+				return types.ErrSwapOutFailed.Wrapf("SwapOutInfo of this gvg(ID: %d) already exist.", gvgID)
+			}
+			swapOutInfo := &types.SwapOutInfo{
+				Id:            gvgID,
```

Comment:
Can we set Id to nil to save space?

Reply:
Remove the ID to save space because i think its useless.